tidb_query_codegen unit test fails

## Bug Report

Run `make dev`

```
     Running target/debug/deps/tidb_query_codegen-03c816ee687b241e
running 11 tests
test rpn_function::tests_normal::test_no_generic_generate_fn_trait ... ok
test rpn_function::tests_normal::test_no_generic_generate_dummy_fn_trait_impl ... ok
test rpn_function::tests_normal::test_generic_generate_fn_trait ... ok
test rpn_function::tests_normal::test_generic_generate_dummy_fn_trait_impl ... ok
test rpn_function::tests_normal::test_no_generic_generate_evaluator ... ok
test rpn_function::tests_normal::test_generic_generate_evaluator ... ok
test rpn_function::tests_normal::test_generic_generate_real_fn_trait_impl ... ok
test rpn_function::tests_normal::test_no_generic_with_extras_generate_real_fn_trait_impl ... ok
test rpn_function::tests_normal::test_no_generic_generate_real_fn_trait_impl ... ok
thread 'rpn_function::tests_normal::test_no_generic_generate_constructor' panicked at 'assertion failed: `(left == right)`
  left: `"pub const fn foo_fn_meta ( ) -> crate :: rpn_expr :: RpnFnMeta { # [ inline ] fn run ( ctx : & mut crate :: expr :: EvalContext , output_rows : usize , args : & [ crate :: rpn_expr :: RpnStackNode < \'_ > ] , extra : & mut crate :: rpn_expr :: RpnFnCallExtra < \'_ > , metadata : & ( dyn std :: any :: Any + Send ) , ) -> crate :: Result < crate :: codec :: data_type :: VectorValue > { use crate :: rpn_expr :: function :: { ArgConstructor , Evaluator , Null } ; < ArgConstructor < Real , _ >> :: new ( 1usize , < ArgConstructor < Int , _ >> :: new ( 0usize , Foo_Evaluator ( std :: marker :: PhantomData ) ) ) . eval ( Null , ctx , output_rows , args , extra , metadata ) } fn init_metadata ( expr : & mut :: tipb :: Expr ) -> Box < dyn std :: any :: Any + Send > { Box :: new ( ( ) ) } fn validate ( expr : & tipb :: Expr ) -> crate :: Result < ( ) > { use crate :: codec :: data_type :: Evaluable ; use crate :: rpn_expr :: function ; function :: validate_expr_return_type ( expr , Decimal :: EVAL_TYPE ) ? ; function :: validate_expr_arguments_eq ( expr , 2usize ) ? ; let children = expr . get_children ( ) ; function :: validate_expr_return_type ( & children [ 0usize ] , Int :: EVAL_TYPE ) ? ; function :: validate_expr_return_type ( & children [ 1usize ] , Real :: EVAL_TYPE ) ? ; Ok ( ( ) ) } crate :: rpn_expr :: RpnFnMeta { name : \"foo\" , metadata_ctor_ptr : init_metadata , validator_ptr : validate , fn_ptr : run , } }"`,
 right: `"pub const fn foo_fn_meta ( ) -> crate :: rpn_expr :: RpnFnMeta { # [ inline ] fn run ( ctx : & mut crate :: expr :: EvalContext , output_rows : usize , args : & [ crate :: rpn_expr :: RpnStackNode < \'_ > ] , extra : & mut crate :: rpn_expr :: RpnFnCallExtra < \'_ > , metadata : & ( dyn std :: any :: Any + Send ) , ) -> crate :: Result < crate :: codec :: data_type :: VectorValue > { use crate :: rpn_expr :: function :: { ArgConstructor , Evaluator , Null } ; < ArgConstructor < Real , _ >> :: new ( 1usize , < ArgConstructor < Int , _ >> :: new ( 0usize , Foo_Evaluator ( std :: marker :: PhantomData ) ) ) . eval ( Null , ctx , output_rows , args , extra , metadata ) } fn init_metadata ( expr : & mut :: tipb :: Expr ) -> Result < Box < dyn std :: any :: Any + Send >> { Ok ( Box :: new ( ( ) ) ) } fn validate ( expr : & tipb :: Expr ) -> crate :: Result < ( ) > { use crate :: codec :: data_type :: Evaluable ; use crate :: rpn_expr :: function ; function :: validate_expr_return_type ( expr , Decimal :: EVAL_TYPE ) ? ; function :: validate_expr_arguments_eq ( expr , 2usize ) ? ; let children = expr . get_children ( ) ; function :: validate_expr_return_type ( & children [ 0usize ] , Int :: EVAL_TYPE ) ? ; function :: validate_expr_return_type ( & children [ 1usize ] , Real :: EVAL_TYPE ) ? ; Ok ( ( ) ) } crate :: rpn_expr :: RpnFnMeta { name : \"foo\" , metadata_ctor_ptr : init_metadata , validator_ptr : validate , fn_ptr : run , } }"`', components/tidb_query_codegen/src/rpn_function.rs:1240:9
stack backtrace:
thread 'rpn_function::tests_normal::test_generic_generate_constructor' panicked at 'assertion failed: `(left == right)`
  left: `"pub const fn foo_fn_meta < A : M , B > ( ) -> crate :: rpn_expr :: RpnFnMeta where B : N < A > { # [ inline ] fn run < A : M , B > ( ctx : & mut crate :: expr :: EvalContext , output_rows : usize , args : & [ crate :: rpn_expr :: RpnStackNode < \'_ > ] , extra : & mut crate :: rpn_expr :: RpnFnCallExtra < \'_ > , metadata : & ( dyn std :: any :: Any + Send ) , ) -> crate :: Result < crate :: codec :: data_type :: VectorValue > where B : N < A > { use crate :: rpn_expr :: function :: { ArgConstructor , Evaluator , Null } ; < ArgConstructor < A :: X , _ >> :: new ( 0usize , Foo_Evaluator :: < A , B > ( std :: marker :: PhantomData ) ) . eval ( Null , ctx , output_rows , args , extra , metadata ) } fn init_metadata < A : M , B > ( expr : & mut :: tipb :: Expr ) -> Box < dyn std :: any :: Any + Send > where B : N < A > { Box :: new ( ( ) ) } fn validate < A : M , B > ( expr : & tipb :: Expr ) -> crate :: Result < ( ) > where B : N < A > { use crate :: codec :: data_type :: Evaluable ; use crate :: rpn_expr :: function ; function :: validate_expr_return_type ( expr , B :: EVAL_TYPE ) ? ; function :: validate_expr_arguments_eq ( expr , 1usize ) ? ; let children = expr . get_children ( ) ; function :: validate_expr_return_type ( & children [ 0usize ] , A :: X :: EVAL_TYPE ) ? ; Ok ( ( ) ) } crate :: rpn_expr :: RpnFnMeta { name : \"foo\" , metadata_ctor_ptr : init_metadata :: < A , B > , validator_ptr : validate :: < A , B > , fn_ptr : run :: < A , B > , } }"`,
 right: `"pub const fn foo_fn_meta < A : M , B > ( ) -> crate :: rpn_expr :: RpnFnMeta where B : N < A > { # [ inline ] fn run < A : M , B > ( ctx : & mut crate :: expr :: EvalContext , output_rows : usize , args : & [ crate :: rpn_expr :: RpnStackNode < \'_ > ] , extra : & mut crate :: rpn_expr :: RpnFnCallExtra < \'_ > , metadata : & ( dyn std :: any :: Any + Send ) , ) -> crate :: Result < crate :: codec :: data_type :: VectorValue > where B : N < A > { use crate :: rpn_expr :: function :: { ArgConstructor , Evaluator , Null } ; < ArgConstructor < A :: X , _ >> :: new ( 0usize , Foo_Evaluator :: < A , B > ( std :: marker :: PhantomData ) ) . eval ( Null , ctx , output_rows , args , extra , metadata ) } fn init_metadata < A : M , B > ( expr : & mut :: tipb :: Expr ) -> Result < Box < dyn std :: any :: Any + Send >> where B : N < A > { Ok ( Box :: new ( ( ) ) ) } fn validate < A : M , B > ( expr : & tipb :: Expr ) -> crate :: Result < ( ) > where B : N < A > { use crate :: codec :: data_type :: Evaluable ; use crate :: rpn_expr :: function ; function :: validate_expr_return_type ( expr , B :: EVAL_TYPE ) ? ; function :: validate_expr_arguments_eq ( expr , 1usize ) ? ; let children = expr . get_children ( ) ; function :: validate_expr_return_type ( & children [ 0usize ] , A :: X :: EVAL_TYPE ) ? ; Ok ( ( ) ) } crate :: rpn_expr :: RpnFnMeta { name : \"foo\" , metadata_ctor_ptr : init_metadata :: < A , B > , validator_ptr : validate :: < A , B > , fn_ptr : run :: < A , B > , } }"`', components/tidb_query_codegen/src/rpn_function.rs:1418:9
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.35/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.35/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:47
   3: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:36
   4: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:200
   5: std::panicking::default_hook
             at src/libstd/panicking.rs:214
   6: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:481
   7: std::panicking::continue_panic_fmt
             at src/libstd/panicking.rs:384
   8: std::panicking::begin_panic_fmt
             at src/libstd/panicking.rs:339
   9: tidb_query_codegen::rpn_function::tests_normal::test_no_generic_generate_constructor
             at components/tidb_query_codegen/src/rpn_function.rs:1240
  10: tidb_query_codegen::rpn_function::tests_normal::test_no_generic_generate_constructor::{{closure}}
             at components/tidb_query_codegen/src/rpn_function.rs:1198
  11: core::ops::function::FnOnce::call_once
             at /rustc/c6e9c76c59e3c10acd63ca9ec157a8894ea1a068/src/libcore/ops/function.rs:227
  12: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
             at /rustc/c6e9c76c59e3c10acd63ca9ec157a8894ea1a068/src/liballoc/boxed.rs:922
  13: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:80
  14: std::panicking::try
             at /rustc/c6e9c76c59e3c10acd63ca9ec157a8894ea1a068/src/libstd/panicking.rs:275
  15: std::panic::catch_unwind
             at /rustc/c6e9c76c59e3c10acd63ca9ec157a8894ea1a068/src/libstd/panic.rs:394
  16: test::run_test::run_test_inner::{{closure}}
             at src/libtest/lib.rs:1413
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.35/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.35/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:47
   3: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:36
   4: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:200
   5: stdtest rpn_function::tests_normal::test_no_generic_generate_constructor ... ::panickingFAILED::
default_hook
             at src/libstd/panicking.rs:214
   6: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:481
   7: std::panicking::continue_panic_fmt
             at src/libstd/panicking.rs:384
   8: std::panicking::begin_panic_fmt
             at src/libstd/panicking.rs:339
   9: tidb_query_codegen::rpn_function::tests_normal::test_generic_generate_constructor
             at components/tidb_query_codegen/src/rpn_function.rs:1418
  10: tidb_query_codegen::rpn_function::tests_normal::test_generic_generate_constructor::{{closure}}
             at components/tidb_query_codegen/src/rpn_function.rs:1370
  11: core::ops::function::FnOnce::call_once
             at /rustc/c6e9c76c59e3c10acd63ca9ec157a8894ea1a068/src/libcore/ops/function.rs:227
  12: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
             at /rustc/c6e9c76c59e3c10acd63ca9ec157a8894ea1a068/src/liballoc/boxed.rs:922
  13: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:80
  14: std::panicking::try
             at /rustc/c6e9c76c59e3c10acd63ca9ec157a8894ea1a068/src/libstd/panicking.rs:275
  15: std::panic::catch_unwind
             at /rustc/c6e9c76c59e3c10acd63ca9ec157a8894ea1a068/src/libstd/panic.rs:394
  16: test::run_test::run_test_inner::{{closure}}
             at src/libtest/lib.rs:1413
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
test rpn_function::tests_normal::test_generic_generate_constructor ... FAILED
failures:
failures:
    rpn_function::tests_normal::test_generic_generate_constructor
    rpn_function::tests_normal::test_no_generic_generate_constructor
test result: FAILED. 9 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out
error: test failed, to rerun pass '-p tidb_query_codegen --lib'
Makefile:217: recipe for target 'test' failed
make[1]: *** [test] Error 101
make[1]: Leaving directory '/home/geekypanda/Softwares/tikv'
Makefile:108: recipe for target 'dev' failed
make: *** [dev] Error 2
```

--- a/components/tidb_query_codegen/src/rpn_function.rs
+++ b/components/tidb_query_codegen/src/rpn_function.rs
@@ -1214,9 +1214,9 @@ mod tests_normal {
                     )
                     .eval(Null, ctx, output_rows, args, extra, metadata)
                 }
-                fn init_metadata(expr: &mut ::tipb::Expr)-> Box<dyn std::any::Any + Send>
+                fn init_metadata(expr: &mut ::tipb::Expr)-> Result<Box<dyn std::any::Any + Send>>
                 {
-                    Box::new(())
+                    Ok(Box::new(()))
                 }
                 fn validate(expr: &tipb::Expr) -> crate::Result<()> {
                     use crate::codec::data_type::Evaluable;
@@ -1389,11 +1389,11 @@ mod tests_normal {
                     <ArgConstructor<A::X, _>>::new(0usize, Foo_Evaluator::<A, B>(std::marker::PhantomData))
                                 .eval(Null, ctx, output_rows, args, extra, metadata)
                 }
-                fn init_metadata <A: M, B> (expr: &mut ::tipb::Expr)-> Box<dyn std::any::Any + Send>
+                fn init_metadata <A: M, B> (expr: &mut ::tipb::Expr)-> Result<Box<dyn std::any::Any + Send>>
                 where
                     B: N<A>
                 {
-                    Box::new(())
+                    Ok(Box::new(()))
                 }
                 fn validate<A: M, B>(expr: &tipb::Expr) -> crate::Result<()>
                 where

